### ISSUE TITLE
Black screen when booting with dynamic graphics mode

### ISSUE THREAD

**Issue by Dave-Swagten** [yusufklncc/Lenovo-Legion-5-Hackintosh#8]
Im sure alot of other people have (had) this problem. I will do my best to find a fix and post it on here. Thank to Yusuf for creating this awesome efi, we can almost have a great hackintosh experience!

**Comment by yusufklncc** [comment]
Thanks mate. I dont have this laptop for now. But i updated last EFI backup when i have this computer. Can you try with this? If issue wont solve i hope you can find problem and fix it.
[EFI.zip](https://github.com/yusufklncc/Lenovo-Legion-5-Hackintosh/files/10094570/EFI.zip)

**Comment by rogeriobrito84** [comment]
I'm also looking for this solution :(

**Comment by Dave-Swagten** [comment]
Hello guys, im happy to tell you that i found a fix. i wil upload the EFI file so that the op can commit to main (If you feel like it :P). **Dont forget to change serial and rom**. Have Fun hackintoshing!

[EFI.zip](https://github.com/yusufklncc/Lenovo-Legion-5-Hackintosh/files/10103767/EFI.zip)
(because of github only allowing us to upload 25mb max, i uploaded a part of the efi. What you have to do now is overwrite the efi from release with this and change serial and rom. after that, boot into your new hackintosh!)

**Comment by yusufklncc** [comment]
Hm, I see, you disabled the external video card with SSDT. When I disabled it with SSDT, there was a fan noise problem. There was no problem when I disabled it from Device Properties. Can you try to delete the # values at the beginning of my patch in Device Properties instead of this SSDT? I'm wondering if it's working. Because SSDT doesn't work for me either.

@Dave-Swagten

**Comment by yusufklncc** [comment]
What did you change then? I didn't see anything.

**Comment by yusufklncc** [comment]
@Dave-Swagten

**Comment by rogeriobrito84** [comment]
Has the problem been resolved?

**Comment by rogeriobrito84** [comment]
I tested this one from @Dave-Swagten , it failed to boot

**Comment by FabriSC** [comment]
you have to copy the efi of @Dave-Swagten with the efi of @yusufklncc and it will work

**Comment by rogeriobrito84** [comment]
updated the SMBIOS, copied the replacing files.
Even so it didn't work!

**Comment by yusufklncc** [comment]
I updated EFI. Try it. It will work.
https://github.com/yusufklncc/Lenovo-Legion-5-Hackintosh/releases

**Comment by yusufklncc** [comment]
@rogeriobrito84

**Comment by rogeriobrito84** [comment]
I updated the SMBIOS, copied the replacing files.
The boot worked, but now it is in the old problem in dynamic mode the black screen.

**Comment by Dave-Swagten** [comment]
### If it still doesnt work, download the original efi and:
### **remove:** 
-device-id: test and AAPL,GfxYTile in PciRoot(0x0)/Pci(0x2,0x0)
-kext: FeatureUnlock.kext
-boot arg: -igfxfw=2
-SSDT-dGPU-Off.aml and Spoof-SSDT.aml

### **Set:**
-Both MinDate and MinVersion to 0
-dpcd-max-link-rate -> 1E000000
-enable-cdclk-frequency-fix -> 01000000

### Make sure to test each:
MacBookPro16,1, or MacBookPro16,3, or MacBookPro16,4.

Also make sure the path of your igpu is set up correctly. For me, this was PciRoot(0x0)/Pci(0x2,0x0).

**Remember to reset nvram on any change you make and generate new valid smbios values!!!**

**Comment by rogeriobrito84** [comment]
Guys, I'm not a very experienced hackintosh user.
I tried to do what nnnnnn asked but in the end it gave an error.

![image](https://user-images.githubusercontent.com/5102484/204632320-0a9aa489-3e14-4b30-bb54-f610d8a9b1a5.png)


I think I'm going to give up :(

**Comment by yusufklncc** [comment]
https://github.com/yusufklncc/Lenovo-Legion-5-Hackintosh/releases/tag/0.8.6
Test it.

**Comment by rogeriobrito84** [comment]
I took the one that @yusufklncc  sent and did the smbios, it got the black screen after about 5 minutes the screen appeared. That's weird!
I formatted the HD started the installation gave the loading time and continued.

![image](https://user-images.githubusercontent.com/5102484/204787656-41242e9b-bf83-4c37-acd1-373b580933d4.png)


After restarting the first time, it stuck on this screen.


![image](https://user-images.githubusercontent.com/5102484/204787820-20cc0e08-4237-4722-9e51-7ce0d162d7a9.png)


Rebooted twice, it goes into setup and forever on that screen.
